app/models/invoice.rb

```diff
@@ -30,4 +30,8 @@ def cancel_previous_invoices
   def display_number
     "#{order.distributor.id}-#{number}"
   end
+
+  def previous_invoice
+    order.invoices.where("id < ?", id).first
```

Comment:
If an order can have more that one previous invoice, we should use ordering to make sure it's the most recent one.

Reply:
I'm intrigued to see that the spec does cover this. But it still seems worth making it explicit here.

Reply:
Hi @dacook 
Invoices are sorted by default as follows:
```ruby
default_scope { order(created_at: :desc) }
```